crates/codegen/runtime/generator/src/parser/grammar/resolver.rs

```diff
@@ -301,16 +282,15 @@ fn resolve_grammar_element(ident: &Identifier, ctx: &mut ResolveCtx<'_>) -> Gram
         // First time resolving a terminal named `ident`
         (None, None) => {
             let named_scanner = match elem {
-                Item::Trivia { item } => Rc::clone(item) as Rc<_>,
-                Item::Fragment { item } => Rc::clone(item) as Rc<_>,
-                Item::Token { item } => Rc::clone(item) as Rc<_>,
                 Item::Keyword { item } => {
                     // Keywords are special scanners and are handled separately
-                    let resolved =
-                        GrammarElement::KeywordScannerDefinition(Rc::clone(item) as Rc<_>);
+                    let resolved = GrammarElement::KeywordScannerDefinition(item as Rc<_>);
                     ctx.resolved.insert(ident.clone(), resolved.clone());
                     return resolved;
                 }
+                Item::Token { item } => item as Rc<_>,
+                Item::Trivia { item } => item as Rc<_>,
+                Item::Fragment { item } => item as Rc<_>,
                 _ => unreachable!("Only terminals can be resolved here"),
```

Comment:
nit: for here and other panics: I suggest adding the provided `ident` to the error message, to make it easier to debug what went wrong.

Reply:
Done in https://github.com/NomicFoundation/slang/pull/1018/commits/2ffeb5e2e25f36af2bb4057b7bc149d50cebc784, I agree with the spirit, however, there's no real value IMO to do that for these `unreachable`s specifically - it's checked above in the match whether the branch is concerned with a parser thunk (so nonterminal) or not; the code does not change often and the assertion seems fairly fundamental (what we consider a non-terminal).

For things like `panic!("Empty error_recovery")` it makes more sense but ideally we should validate or model that better in the DSL v2, instead.